validator.py

```diff
@@ -793,45 +759,55 @@ def character_matching(self):
         # get all locations that have character ids 
         allowed_loc_0 = "state.characters[].id".split('.')
         allowed_loc_other = "scenes[].state.characters[].id".split('.')
-        locations_0 = self.property_meets_conditions(allowed_loc_0, copy.deepcopy(self.loaded_yaml))
-        locations_other = self.property_meets_conditions(allowed_loc_other, copy.deepcopy(self.loaded_yaml))
-        allowed_vals = {0: []}
+        data = copy.deepcopy(self.loaded_yaml)
+        locations_0 = self.property_meets_conditions(allowed_loc_0, data)
+        locations_other = self.property_meets_conditions(allowed_loc_other, data)
+        scenes = data['scenes']
+        first_scene = data['first_scene'] if 'first_scene' in data else None
+        if first_scene is None:
+            first_scene = scenes[0]['id']
+        allowed_vals = {}
+        allowed_vals[first_scene] = []
         # get all allowed values, organizing by the scene index where those values will be allowed
         for l in locations_0:
             loc = l.split('.')
-            val = self.get_value_at_key(loc, copy.deepcopy(self.loaded_yaml))
+            val = self.get_value_at_key(loc, data)
             if val is not None:
-                allowed_vals[0].append(val)   
+                allowed_vals[first_scene].append(val)   
         for l in locations_other:
             ind = int(l.split('cenes[')[1].split(']')[0])
```

Comment:
is `l.split('cenes[')` a typo?

Reply:
It is not. Just catches if it's "Scenes" or "scenes" or something else similar to that

Reply:
It appears to be working so I guess not, I'm just confused on how this line works

Reply:
Actually, I might have done it that way because sometimes it can be confusing how python splits if what you're looking for is the very beginning of the string, so I wanted it to be ['s', 'whatever's after cenes[']

Reply:
> I'm just confused on how this line works

This happens to me whenever the number of punctation characters exceeds the number of alphabetic characters in a line of code.